pydm/widgets/base.py

```diff
@@ -1043,3 +1044,8 @@ def check_enable_state(self):
                 tooltip += "Access denied by Channel Access Security."
         self.setToolTip(tooltip)
         self.setEnabled(status)
+
+    def deleteLater(self):
+        """Close widget connections before deleting"""
+        close_widget_connections(self)
```

Comment:
I worry a little bit about the possibility that `close_widget_connections` throws an error or something, and effectively prevents the widget from actually getting deleted.

I think it may be preferable to actually call `deleteLater` _first_, then this.